internal/db/schema/migrations/oss/postgres/101/02_app_token_global.up.sql

```diff
@@ -0,0 +1,220 @@
+    create_time wt_timestamp,
+    constraint app_token_permission_global_grant_scope_private_id_uq
+      unique(grant_scope, private_id)
+  );
+  comment on table app_token_permission_global is
+    'app_token_permission_global contains global scope-specific permissions for app tokens.';
+
+  -- Create index on app_token_id for better query performance
+  create index app_token_permission_global_app_token_id_idx on app_token_permission_global (app_token_id);
+
+  -- Add triggers for app_token_permission_global
+  create trigger default_create_time_column before insert on app_token_permission_global
+    for each row execute procedure default_create_time();
+
+  create trigger immutable_columns before update on app_token_permission_global
+    for each row execute procedure immutable_columns('app_token_id', 'scope_id', 'label', 'grant_scope', 'grant_this_scope', 'create_time');
+
+  create trigger insert_app_token_permission_subtype before insert on app_token_permission_global
+    for each row execute procedure insert_app_token_permission_subtype();
+
+  create table app_token_permission_global_individual_org_grant_scope (
+    permission_id wt_private_id
+      constraint app_token_permission_global_fkey
+      references app_token_permission_global(private_id)
+        on delete cascade
+        on update cascade,
+    scope_id wt_scope_id not null,
+    -- grant_scope is used for constraint checking.
+    -- This restricts the grant_scope to be 'individual'
+    -- and since it is also a foreign key to the app_token_permission_global
+    -- grant_scope, it ensures that app_token_permission_global is set to 'individual'
+    -- if this table is populated for the corresponding permission.
+    grant_scope text not null
+       constraint only_individual_grant_scope_allowed
+         check(
+          grant_scope = 'individual'
+        ),
+    create_time wt_timestamp,
+    constraint app_token_permission_global_grant_scope_fkey
+      foreign key (grant_scope, permission_id)
+      references app_token_permission_global(grant_scope, private_id)
+      on delete cascade
+      on update cascade,
+    primary key(permission_id, scope_id)
+  );
+  comment on table app_token_permission_global_individual_org_grant_scope is
+    'app_token_permission_global_individual_org_grant_scope is a list of individually granted org scope to global app token permissions with grant_scope of individual.';
+
+  create or replace function validate_global_permission_org_scope() returns trigger
+  as $$
+  begin
+    perform
+       from iam_scope_org
+      where iam_scope_org.scope_id = new.scope_id;
+    if not found then
+      raise exception 'org scope_id % not found', new.scope_id;
+    end if;
+    return new;
+  end;
+  $$ language plpgsql;
+    comment on function validate_global_permission_org_scope() is
+      'validate_global_permission_org_scope is used to enforced that scope ID added to app_token_permission_global_individual_project_grant_scope'
```

Comment:
Small grammar nitpick -

```suggestion
      'validate_global_permission_org_scope is used to enforce that scope ID added to app_token_permission_global_individual_project_grant_scope'
```